.github/workflows/e2e.yaml

```diff
@@ -97,10 +104,18 @@ jobs:
       - id: generate-cluster-name
         name: generate cluster name
         run: |
-          CLUSTER_NAME="$(echo ${{ inputs.suite }}-"$RANDOM$RANDOM" | awk '{print tolower($0)}' | tr / -)"
-          echo Using cluster name "$CLUSTER_NAME"
+          CLUSTER_NAME='${{ inputs.cluster_name }}'
+          if [[ '${{ inputs.cluster_name }}' == '' ]]; then 
+            if [[ '${{ inputs.workflow_trigger }}' == 'soak' ]]; then 
+              CLUSTER_NAME=$(echo soak-periodic-$RANDOM$RANDOM | awk '{print tolower($0)}' | tr / -)
+            else 
+              CLUSTER_NAME=$(echo ${{ inputs.suite }}-$RANDOM$RANDOM | awk '{print tolower($0)}' | tr / -)
+            fi 
+          fi
```

Comment:
nit: I think the way you did it in action.yaml looks cleaner, but up to you. https://github.com/aws/karpenter/pull/4996/files#diff-4eb536776cca4973b074324c28a1f76a816f488a2bd4c1de81f8b1585fab402bL23-L27

```
if a & b  { 
  set cluster name
} 
elif a & !b {
   set cluster name
}
```